peps/pep-0757.rst

```diff
@@ -263,15 +263,25 @@ Code::
             const PyLongLayout* layout = PyLong_GetNativeLayout();
             static PyLong_DigitArray long_export;
 
-            PyLong_AsDigitArray(obj, &long_export);
-            mpz_import(z, long_export.ndigits, layout->endian,
-                       layout->digit_size, layout->digits_order,
-                       layout->digit_size*8 - layout->bits_per_digit,
-                       long_export.digits);
+            PyLong_Export(obj, &long_export);
+            if (long_export.digits) {
+                mpz_import(z, long_export.ndigits, layout->digits_order,
+                           layout->digit_size, layout->endian,
+                           layout->digit_size*8 - layout->bits_per_digit,
+                           long_export.digits);
+            }
+            else {
+                if (long_export.negative) {
+                    long_export.value = -long_export.value;
```

Comment:
It can overflow if value is INT64_MIN. I would suggest to not use negative if digits is NULL.

Reply:
Indeed :(  Unfortunately, to unpack from twos complement we need a temporary mpz.

Maybe store absolute value in ``uint64_t value``?

Reply:
Do you mean to use value+negative in this case? I would be fine with that.